gum/backend-windows/gumprocess-windows.c

```diff
@@ -305,13 +305,70 @@ _gum_process_enumerate_ranges (GumPageProtection prot,
   }
 }
 
+static gboolean
+gum_process_enumerate_heap_ranges (HANDLE heap, GumFoundMallocRangeFunc func,
+                                   gpointer user_data)
+{
+  PROCESS_HEAP_ENTRY entry;
+  GumMallocRangeDetails details;
+  GumMemoryRange gum_range;
+  gboolean carry_on, unlock_heap;
+
+  // HeapLock may fail but it doesn't seem to have any real consequences...
+  unlock_heap = HeapLock (heap);
+
+  details.range = &gum_range;
+  carry_on = TRUE;
+  entry.lpData = NULL;
+  while (carry_on && HeapWalk (heap, &entry)) {
+    if ((entry.wFlags & PROCESS_HEAP_ENTRY_BUSY) != 0) {
+      gum_range.base_address = entry.lpData;
+      gum_range.size = entry.cbData;
+      carry_on = func (&details, user_data);
+    }
+  }
+  if (unlock_heap)
```

Comment:
Let's add a blank line before this `if` so it reads better. (Structure of this function is: lock, do stuff, unlock. Three logical groups, which should have a blank line between each of them.)